src/items/functions.md

```diff
@@ -93,52 +93,6 @@ sufficient context to determine the type parameters. For example,
 
 [path]: paths.html
 
-## Diverging functions
```

Comment:
Why remove this? The information about what constructs produce the `!` return type remains useful. Also, the new section below doesn't use "diverges", "diverge", or "divergent", though we make use of "diverges" in the section about `loop {}`s.

I say just keep this section in tact, with adjustments to account for new thinking about the Never Type.

Reply:
There's not really a good reason to have information on divergence in the function items page. Perhaps mention in the function call and method call operators that calling a function that returns `!` diverges? Divergence sort of needs its own section in the Expressions page itself, I think? I haven't thought too much about whether it's a property of expressions, statements, or both? And that can be done separately from this.

It would be nice to pull the example into the Never type docs though.

Reply:
I agree with @Havvy here. Divergence isn't specifically a property of functions, it's its own thing entirely and really needs its own section. I don't fully understand the semantics of it though, so I'm not confident that I'd be able to write that section.

Reply:
Yeah, put the information where ever you think it fits best, but I don't want to _lose_ the information.

Reply:
Maybe someone like @arielb1 can help here. Is this an accurate-enough description of type-checking w.r.t divergence and `!`?

* Expressions of type `!` cause everything in the computation graph after the `!` to be marked unreachable.
* If a block ends in a statement, and the end of the block is unreachable, then the block has type `!`.

I know it's not exactly implemented like this in the compiler, but have I left out any relevant caveats? Do you think the reference needs a section on divergence separate from `!`?

Reply:
I'll make sure that divergence information doesn't get lost before the next beta, even if this PR loses it.